CRM/Civicase/Form/Report/BaseExtendedReport.php

```diff
@@ -811,6 +811,47 @@ private function getFieldOptions(array $spec, $fieldName) {
     );
   }
 
+  /**
+   * Build custom data from clause.
+   *
+   * Overridden to support custom data for multiple entities of the same type.
+   */
+  public function extendedCustomDataFrom() {
+    foreach ($this->getMetadataByType('metadata') as $prop) {
+      $table = $prop['table_name'];
+      if (empty($prop['extends']) || !$this->isCustomTableSelected($table)) {
+        continue;
+      }
+
+      $baseJoin = CRM_Utils_Array::value($prop['extends'], $this->_customGroupExtendsJoin, "{$this->_aliases[$prop['extends_table']]}.id");
+
+      $customJoin = is_array($this->_customGroupJoin) ? $this->_customGroupJoin[$table] : $this->_customGroupJoin;
+      $tableKey = CRM_Utils_Array::value('prefix', $prop) . $prop['table_name'];
+      if (!stristr($this->_from, ' ' . $this->_aliases[$tableKey] . ' ')) {
```

Comment:
I thought the aliases were unique for each table and this issue should not occur? 
The way it is now won't there be a situation where a from condition is repeated twice based on the spaces wrapped at the beginning and end of the alias?

Reply:
well each custom value table that belongs to contact is repeated with a role alias at the beginning so for example if the custom value table is `test` and there are two roles `solicitor` and `criminal_solicitor` so there will be two joins by the name of criminal_solicitor_test and solicitor_test but before joining this check was there to stop duplicating the join but this will error in this example if criminal_solicitor_test has already been join so this check will not let solicitor_test to be joined as the from already has that string so I changed the check to look for complete alias in from